stan/math/prim/fun/eigenvalues.hpp

```diff
@@ -7,14 +7,28 @@
 namespace stan {
 namespace math {
 
-template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr>
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+          require_not_vt_complex<EigMat>* = nullptr>
 inline auto eigenvalues(const EigMat& m) {
   using PlainMat = plain_type_t<EigMat>;
   const PlainMat& m_eval = m;
   check_nonzero_size("eigenvalues", "m", m_eval);
```

Comment:
Just wanted to double-check: if there's a 0x0 matrix provided, we're throwing instead of returning back a 0-vector of eigenvalues?

Reply:
Yes. This is the behavior of the real-valued versions added in https://github.com/stan-dev/math/pull/1838, but if you think we should return an empty vector instead we could make the change to both signatures

Reply:
The functions `eigen[vectors|values]_sym` both have this behavior as well so I think it's best to maintain it for this PR